src/precompiles/native.rs

```diff
@@ -0,0 +1,267 @@
+const MAX_ACCOUNT_ID_LEN: u64 = 64;
+
+/// Returns `true` if the given account ID is valid and `false` otherwise.
+///
+/// Taken from near-sdk-rs:
+/// (https://github.com/near/near-sdk-rs/blob/42f62384c3acd024829501ee86e480917da03896/near-sdk/src/environment/env.rs#L816-L843)
+pub fn is_valid_account_id(account_id: &[u8]) -> bool {
+    if (account_id.len() as u64) < MIN_ACCOUNT_ID_LEN
+        || (account_id.len() as u64) > MAX_ACCOUNT_ID_LEN
+    {
+        return false;
+    }
+
+    // NOTE: We don't want to use Regex here, because it requires extra time to compile it.
+    // The valid account ID regex is /^(([a-z\d]+[-_])*[a-z\d]+\.)*([a-z\d]+[-_])*[a-z\d]+$/
+    // Instead the implementation is based on the previous character checks.
+
+    // We can safely assume that last char was a separator.
+    let mut last_char_is_separator = true;
+
+    for c in account_id {
+        let current_char_is_separator = match *c {
+            b'a'..=b'z' | b'0'..=b'9' => false,
+            b'-' | b'_' | b'.' => true,
+            _ => return false,
+        };
+        if current_char_is_separator && last_char_is_separator {
+            return false;
+        }
+        last_char_is_separator = current_char_is_separator;
+    }
+    // The account can't end as separator.
+    !last_char_is_separator
+}
+
+pub struct ExitToNear; //TransferEthToNear
+
+impl Precompile for ExitToNear {
+    fn required_gas(_input: &[u8]) -> Result<u64, ExitError> {
+        Ok(costs::EXIT_TO_NEAR_GAS)
+    }
+
+    #[cfg(not(feature = "contract"))]
+    fn run(input: &[u8], target_gas: u64, context: &Context) -> PrecompileResult {
+        if Self::required_gas(input)? > target_gas {
+            return Err(ExitError::OutOfGas);
+        }
+
+        Ok((ExitSucceed::Returned, Vec::new(), 0))
+    }
+
+    #[cfg(feature = "contract")]
+    fn run(input: &[u8], target_gas: u64, context: &Context) -> PrecompileResult {
+        if Self::required_gas(input)? > target_gas {
+            return Err(ExitError::OutOfGas);
+        }
+
+        let (nep141_address, args) = if context.apparent_value != U256::from(0) {
+            // ETH transfer
+            //
+            // Input slice format:
+            //      recipient_account_id (bytes) - the NEAR recipient account which will receive NEP-141 ETH tokens
+
+            if is_valid_account_id(input) {
+                (
+                    String::from_utf8(crate::sdk::current_account_id()).unwrap(),
+                    crate::prelude::format!(
+                        r#"{{"receiver_id": "{}", "amount": "{}", "memo": null}}"#,
+                        String::from_utf8(input.to_vec()).unwrap(),
+                        context.apparent_value.as_u128()
+                    ),
+                )
+            } else {
+                // TODO: Which is the expected result in case of failure
+                //  Stopped vs Returned vs Suicided
+                return Ok((ExitSucceed::Stopped, Vec::new(), 0));
```

Comment:
What is the expected PrecompileResult if it should fail?

Should we return a flag instead `1` on success, and `0` on failure?
```
return Ok((ExitSucceed::Returned, [flag].to_vec(), 0));
```

Reply:
From what I remember using `sha` precompile, when you called it, the result is stored in the range from [0, 32]
However, it also returns a boolean value, whether it succeeded or not. How can I return this boolean value from here?

Reply:
I think we should use `ExitError` for that purpose